### PR TITLE
Don't load all posts until after the first render

### LINKED ISSUE #19: First load is slow
See if it is waiting for all the posts to load. It shouldn't be, but it might be.

### PR DESCRIPTION
**Pull request by TehShrike**
Fixes #19

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -123,24 +140,38 @@ function scan(post, util, state, thisPostChanged) {
...
 	// Create embedded contexts
-	;(state.filenameUuidsMap[post.filename] || []).filter(function (uuid) {
-		return thisPostChanged || !util.partialExists(uuid)
-	}).forEach(function (uuid) {
-		var templateArgs = state.uuidArgumentsMap[uuid]
-		var partialData = extend(post.metadata, templateArgs)
-		util.resetPartial(uuid, makePartialString(post.filename, partialData))
-	})
+	var uuids = state.filenameUuidsMap[post.filename]
+	if (state.filenameUuidsMap[post.filename]) {
```
Comment (ArtskydJ) [inline index.js]: Would it be more clear if this was `if (uuids) {`?

File: package.json [hunk 2]
```diff
@@ -33,8 +34,9 @@
   },
   "devDependencies": {
     "level-mem": "^0.18.0",
-    "noddity-butler": "^2.1.1",
+    "noddity-butler": "~2.3.0",
     "noddity-linkifier": "^2.1.0",
-    "tape": "^4.2.0"
+    "tape": "^4.2.0",
```
Comment (ArtskydJ) [inline package.json]: Should `tape` be removed?

File: index.js (hunk 1 above)
Comment (TehShrike) [inline index.js]: oh yeah, I totally meant to do that

File: package.json (hunk 2 above)
Comment (TehShrike) [inline package.json]: nah, `tape-catch` doesn't specify a dependency on `tape`, you have to do that yourself.  It just requires whatever version you have installed.

**Comment by ArtskydJ** [comment]
lgtm :+1:

**Pull request merged**